apps/src/p5lab/AnimationPicker/AnimationPicker.jsx

```diff
@@ -75,23 +93,60 @@ class AnimationPicker extends React.Component {
     } else if (this.props.uploadInProgress) {
       return <h1 style={styles.title}>{msg.animationPicker_uploading()}</h1>;
     }
+
+    // Update text depending on the context of the animation picker
+    let imageText;
+    switch (this.props.pickerType) {
+      case PICKER_TYPE.spritelab:
+        imageText = 'costumes';
+        break;
+      case PICKER_TYPE.gamelab:
+        imageText = 'animations';
+        break;
+      default:
+        imageText = 'backgrounds';
+    }
+
     return (
-      <AnimationPickerBody
-        is13Plus={this.props.is13Plus}
-        onDrawYourOwnClick={this.props.onPickNewAnimation}
-        onPickLibraryAnimation={this.props.onPickLibraryAnimation}
-        onUploadClick={this.onUploadClick}
-        onAnimationSelectionComplete={this.props.onAnimationSelectionComplete}
-        playAnimations={this.props.playAnimations}
-        libraryManifest={this.props.libraryManifest}
-        hideUploadOption={this.props.hideUploadOption}
-        hideAnimationNames={this.props.hideAnimationNames}
-        navigable={this.props.navigable}
-        defaultQuery={this.props.defaultQuery}
-        hideBackgrounds={this.props.hideBackgrounds}
-        canDraw={this.props.canDraw}
-        selectedAnimations={this.props.selectedAnimations}
-      />
+      <div>
+        <AnimationPickerBody
+          is13Plus={this.props.is13Plus}
+          onDrawYourOwnClick={this.props.onPickNewAnimation}
+          onPickLibraryAnimation={this.props.onPickLibraryAnimation}
+          onUploadClick={this.onUploadClick}
+          onAnimationSelectionComplete={this.props.onAnimationSelectionComplete}
+          playAnimations={this.props.playAnimations}
+          libraryManifest={this.props.libraryManifest}
+          hideUploadOption={this.props.hideUploadOption}
+          hideAnimationNames={this.props.hideAnimationNames}
+          navigable={this.props.navigable}
+          defaultQuery={this.props.defaultQuery}
+          hideBackgrounds={this.props.hideBackgrounds}
+          canDraw={this.props.canDraw}
+          selectedAnimations={this.props.selectedAnimations}
+        />
+        {this.state.exitingDialog && (
+          <BaseDialog isOpen hideCloseButton={true} style={styles.dialog}>
+            <h3>{msg.animationPicker_leaveSelectionTitle()}</h3>
+            <p>{msg.animationPicker_leaveSelectionText({imageText})}</p>
+
+            <DialogFooter rightAlign={true}>
+              <Button
+                text={msg.animationPicker_discardSelection()}
+                onClick={() => {
+                  this.props.onClose();
+                  this.setState({exitingDialog: false});
+                }}
+                color={'gray'}
```

Comment:
nit: this isn't a javascript expression, so it can just be a string like `color="gray"`

Reply:
Oh dang, my IDE defaults to add these! I gotta find a way to fix the default.